Adding setting to only display the label for the selected granularity

KristofferBerge opened the pull request:
Adding setting to hide all labels for higher granularities than the one selected. This saves a lot of vertical space.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/powerbi-visuals-timeline?pullRequest=22) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/powerbi-visuals-timeline?pullRequest=22) before we can accept your contribution.<br/><br/>:x: KristofferBerge [sign now](https://cla.opensource.microsoft.com/Microsoft/powerbi-visuals-timeline?pullRequest=22)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/powerbi-visuals-timeline?pullRequest=22) it.</sub>

savvinsergey commented on src/visual.ts:
```diff
@@ -983,43 +984,60 @@ module powerbi.extensibility.visual {
+                    this.yearLabelsSelection,
+                    yPos,
+                    granularityType === 0);
+                if (timelineSettings.labels.displayAll)
```
Please use - If() { } construction. And I guess you can move this line to before call of rendecCell  function(line 1042). This code is repeated

savvinsergey commented on capabilities.json:
```diff
@@ -313,6 +313,13 @@
                 },
+                "displayAll": {
+                    "displayName": "Display all",
+                    "displayNameKey": "Visual_DisplayAll",
```
Please add this key to stringResources\en-US\resources.resjson

savvinsergey commented on src/visual.ts:
```diff
@@ -555,9 +555,10 @@ module powerbi.extensibility.visual {
             labelSize = fromPointToPixel(labelsSettings.textSize);
 
             if (labelsSettings.show) {
+                let granularityOffset = labelsSettings.displayAll ? granularityType + 1 : 1;
```
Please specify variable type

savvinsergey commented on capabilities.json:
```diff
@@ -313,6 +313,13 @@
+                    "displayName": "Display all",
+                    "displayNameKey": "Visual_DisplayAll",
+                    "type": {
+                        "bool": true
```
Please do not use "true" value by default for your setting cause this can affect to already created users reports

savvinsergey commented on src/settings.ts:
```diff
@@ -53,6 +53,7 @@ module powerbi.extensibility.visual.settings {
 
     export class LabelsSettings {
         show: boolean = true;
+        displayAll: boolean = true;
```
Please do not use "true" value by default for your setting cause this can affect to already created users reports

KristofferBerge commented:
Thanks for the feedback guys. The display all setting should be set to true by default to not change the current behavior.

I've corrected the linting errors and added a unit test to control that the setting has the desired effect, but could maybe be more thorough if I could read the current granularity from the scope of the test and iterate over the different granularities. Any tips on how to do this? Or is the current unit test sufficient?

savvinsergey commented:
> Or is the current unit test sufficient?
I guess that is enought

Please also increase version of this visual to 1.5.0 (pbiviz.json, package.json) and update CHANGELOG file
